### ISSUE TITLE
Dist Debian: missing dependenccies in scylla-server package

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#494]
After building the debian package

Trying to install on the machine it was built - is succesfull

Trying to install on a machine that it was not used to build - it fails

```
ubuntu@ip-10-97-205-120:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
(Reading database ... 51144 files and directories currently installed.)
Preparing to unpack scylla-server_0.10-1_amd64.deb ...
Unpacking scylla-server (0.10-1) over (0.10-1) ...
dpkg: dependency problems prevent configuration of scylla-server:
 scylla-server depends on libaio1 (>= 0.3.93); however:
  Package libaio1 is not installed.
 scylla-server depends on libboost-filesystem1.55.0; however:
  Package libboost-filesystem1.55.0 is not installed.
 scylla-server depends on libboost-program-options1.55.0; however:
  Package libboost-program-options1.55.0 is not installed.
 scylla-server depends on libboost-system1.55.0; however:
  Package libboost-system1.55.0 is not installed.
 scylla-server depends on libboost-test1.55.0; however:
  Package libboost-test1.55.0 is not installed.
 scylla-server depends on libboost-thread1.55.0; however:
  Package libboost-thread1.55.0 is not installed.
 scylla-server depends on libcrypto++9; however:
  Package libcrypto++9 is not installed.
 scylla-server depends on libjsoncpp0 (>= 0.6.0~rc2); however:
  Package libjsoncpp0 is not installed.
 scylla-server depends on liblz4-1 (>= 0.0~r113); however:
  Package liblz4-1 is not installed.
 scylla-server depen
dpkg: error processing package scylla-server (--install):
 dependency problems - leaving unconfigured
Processing triggers for ureadahead (0.100.0-16) ...
Errors were encountered while processing:
 scylla-server
```

after executing apt-get -f install

```
ubuntu@ip-10-97-205-120:~$ sudo apt-get -f install
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
Use 'apt-get autoremove' to remove them.
The following extra packages will be installed:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
Suggested packages:
  libhugetlbfs-tests
The following packages will be REMOVED:
  scylla-server
The following NEW packages will be installed:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
0 upgraded, 13 newly installed, 1 to remove and 125 not upgraded.
1 not fully installed or removed.
Need to get 1,696 kB of archives.
After this operation, 13.8 MB disk space will be freed.
Do you want to continue? [Y/n] y
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/main libaio1 amd64 0.3.109-4 [6,364 B]
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-system1.55.0 amd64 1.55.0-1 [10.6 kB]
Get:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-filesystem1.55.0 amd64 1.55.0-1 [34.2 kB]
Get:4 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-program-options1.55.0 amd64 1.55.0-1 [116 kB]
Get:5 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-test1.55.0 amd64 1.55.0-1 [179 kB]
Get:6 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-thread1.55.0 amd64 1.55.0-1 [27.3 kB]
Get:7 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libyaml-cpp0.5 amd64 0.5.1-1 [207 kB]
Get:8 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libjsoncpp0 amd64 0.6.0~rc2-3ubuntu1 [56.5 kB]
Get:9 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe liblz4-1 amd64 0.0~r114-2ubuntu1 [16.1 kB]
Get:10 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty-updates/universe libcrypto++9 amd64 5.6.1-6+deb8u1build0.14.04.1 [929 kB]
Get:11 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libhugetlbfs0 amd64 2.17-0ubuntu2 [55.4 kB]
Get:12 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe hugepages amd64 2.17-0ubuntu2 [48.4 kB]
Get:13 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/main libsnappy1 amd64 1.1.0-1ubuntu1 [11.2 kB]
Fetched 1,696 kB in 0s (22.5 MB/s)       
(Reading database ... 51144 files and directories currently installed.)
Removing scylla-server (0.10-1) ...
Selecting previously unselected package libaio1:amd64.
(Reading database ... 51123 files and directories currently installed.)
Preparing to unpack .../libaio1_0.3.109-4_amd64.deb ...
Unpacking libaio1:amd64 (0.3.109-4) ...
Selecting previously unselected package libboost-system1.55.0:amd64.
Preparing to unpack .../libboost-system1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-system1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-filesystem1.55.0:amd64.
Preparing to unpack .../libboost-filesystem1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-filesystem1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-program-options1.55.0:amd64.
Preparing to unpack .../libboost-program-options1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-program-options1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-test1.55.0:amd64.
Preparing to unpack .../libboost-test1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-test1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-thread1.55.0:amd64.
Preparing to unpack .../libboost-thread1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-thread1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libyaml-cpp0.5:amd64.
Preparing to unpack .../libyaml-cpp0.5_0.5.1-1_amd64.deb ...
Unpacking libyaml-cpp0.5:amd64 (0.5.1-1) ...
Selecting previously unselected package libjsoncpp0:amd64.
Preparing to unpack .../libjsoncpp0_0.6.0~rc2-3ubuntu1_amd64.deb ...
Unpacking libjsoncpp0:amd64 (0.6.0~rc2-3ubuntu1) ...
Selecting previously unselected package liblz4-1:amd64.
Preparing to unpack .../liblz4-1_0.0~r114-2ubuntu1_amd64.deb ...
Unpacking liblz4-1:amd64 (0.0~r114-2ubuntu1) ...
Selecting previously unselected package libcrypto++9.
Preparing to unpack .../libcrypto++9_5.6.1-6+deb8u1build0.14.04.1_amd64.deb ...
Unpacking libcrypto++9 (5.6.1-6+deb8u1build0.14.04.1) ...
Selecting previously unselected package libhugetlbfs0.
Preparing to unpack .../libhugetlbfs0_2.17-0ubuntu2_amd64.deb ...
Unpacking libhugetlbfs0 (2.17-0ubuntu2) ...
Selecting previously unselected package hugepages.
Preparing to unpack .../hugepages_2.17-0ubuntu2_amd64.deb ...
Unpacking hugepages (2.17-0ubuntu2) ...
Selecting previously unselected package libsnappy1.
Preparing to unpack .../libsnappy1_1.1.0-1ubuntu1_amd64.deb ...
Unpacking libsnappy1 (1.1.0-1ubuntu1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up libaio1:amd64 (0.3.109-4) ...
Setting up libboost-system1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-filesystem1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-program-options1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-test1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-thread1.55.0:amd64 (1.55.0-1) ...
Setting up libyaml-cpp0.5:amd64 (0.5.1-1) ...
Setting up libjsoncpp0:amd64 (0.6.0~rc2-3ubuntu1) ...
Setting up liblz4-1:amd64 (0.0~r114-2ubuntu1) ...
Setting up libcrypto++9 (5.6.1-6+deb8u1build0.14.04.1) ...
Setting up libhugetlbfs0 (2.17-0ubuntu2) ...
Setting up hugepages (2.17-0ubuntu2) ...
Setting up libsnappy1 (1.1.0-1ubuntu1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
```

some of the dependencies are install yet this is not enough - we are still missing some items

```
ubuntu@ip-10-97-205-120:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
Selecting previously unselected package scylla-server.
(Reading database ... 51215 files and directories currently installed.)
Preparing to unpack scylla-server_0.10-1_amd64.deb ...
Unpacking scylla-server (0.10-1) ...
dpkg: dependency problems prevent configuration of scylla-server:
 scylla-server depends on libstdc++6 (>= 4.9); however:
  Version of libstdc++6:amd64 on system is 4.8.2-19ubuntu1.
 scylla-server depends on libthrift0; however:
  Package libthrift0 is not installed.

```

trying to fix the two above by using
```
sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
sudo apt-get -y update
```

and then apt-get -f install

does not work

**Comment by slivne** [comment]
Using 

I created the RPM in a single run

tried installing the deb package on a clean server and failed with 

```
ubuntu@ip-10-230-218-205:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
Selecting previously unselected package scylla-server.
(Reading database ... 62812 files and directories currently installed.)
Preparing to unpack scylla-server_0.10-1_amd64.deb ...
Unpacking scylla-server (0.10-1) ...
dpkg: dependency problems prevent configuration of scylla-server:
 scylla-server depends on libthrift0; however:
  Package libthrift0 is not installed.

dpkg: error processing package scylla-server (--install):
 dependency problems - leaving unconfigured
Processing triggers for ureadahead (0.100.0-16) ...
Errors were encountered while processing:
 scylla-server
```

so thrift is missing

here is the complete log

```
ubuntu@ip-10-230-218-205:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
Selecting previously unselected package scylla-server.
(Reading database ... 62717 files and directories currently installed.)
Preparing to unpack scylla-server_0.10-1_amd64.deb ...
Unpacking scylla-server (0.10-1) ...
dpkg: dependency problems prevent configuration of scylla-server:
 scylla-server depends on libaio1 (>= 0.3.93); however:
  Package libaio1 is not installed.
 scylla-server depends on libboost-filesystem1.55.0; however:
  Package libboost-filesystem1.55.0 is not installed.
 scylla-server depends on libboost-program-options1.55.0; however:
  Package libboost-program-options1.55.0 is not installed.
 scylla-server depends on libboost-system1.55.0; however:
  Package libboost-system1.55.0 is not installed.
 scylla-server depends on libboost-test1.55.0; however:
  Package libboost-test1.55.0 is not installed.
 scylla-server depends on libboost-thread1.55.0; however:
  Package libboost-thread1.55.0 is not installed.
 scylla-server depends on libcrypto++9; however:
  Package libcrypto++9 is not installed.
 scylla-server depends on libjsoncpp0 (>= 0.6.0~rc2); however:
  Package libjsoncpp0 is not installed.
 scylla-server depends on liblz4-1 (>= 0.0~r113); however:
  Package liblz4-1 is not installed.
 scylla-server depen
dpkg: error processing package scylla-server (--install):
 dependency problems - leaving unconfigured
Processing triggers for ureadahead (0.100.0-16) ...
Errors were encountered while processing:
 scylla-server
ubuntu@ip-10-230-218-205:~$ sudo dpkg ^C scylla-server_0.10-1_amd64.deb 
ubuntu@ip-10-230-218-205:~$ auso apt-get -f install
auso: command not found
ubuntu@ip-10-230-218-205:~$ sudo  apt-get -f install
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
Use 'apt-get autoremove' to remove them.
The following extra packages will be installed:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
Suggested packages:
  libhugetlbfs-tests
The following packages will be REMOVED:
  scylla-server
The following NEW packages will be installed:
  hugepages libaio1 libboost-filesystem1.55.0 libboost-program-options1.55.0
  libboost-system1.55.0 libboost-test1.55.0 libboost-thread1.55.0 libcrypto++9
  libhugetlbfs0 libjsoncpp0 liblz4-1 libsnappy1 libyaml-cpp0.5
0 upgraded, 13 newly installed, 1 to remove and 127 not upgraded.
1 not fully installed or removed.
Need to get 1,696 kB of archives.
After this operation, 14.6 MB disk space will be freed.
Do you want to continue? [Y/n] y
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/main libaio1 amd64 0.3.109-4 [6,364 B]
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-system1.55.0 amd64 1.55.0-1 [10.6 kB]
Get:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-filesystem1.55.0 amd64 1.55.0-1 [34.2 kB]
Get:4 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-program-options1.55.0 amd64 1.55.0-1 [116 kB]
Get:5 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-test1.55.0 amd64 1.55.0-1 [179 kB]
Get:6 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libboost-thread1.55.0 amd64 1.55.0-1 [27.3 kB]
Get:7 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libyaml-cpp0.5 amd64 0.5.1-1 [207 kB]
Get:8 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libjsoncpp0 amd64 0.6.0~rc2-3ubuntu1 [56.5 kB]
Get:9 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe liblz4-1 amd64 0.0~r114-2ubuntu1 [16.1 kB]
Get:10 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty-updates/universe libcrypto++9 amd64 5.6.1-6+deb8u1build0.14.04.1 [929 kB]
Get:11 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe libhugetlbfs0 amd64 2.17-0ubuntu2 [55.4 kB]
Get:12 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/universe hugepages amd64 2.17-0ubuntu2 [48.4 kB]
Get:13 http://us-east-1.ec2.archive.ubuntu.com/ubuntu/ trusty/main libsnappy1 amd64 1.1.0-1ubuntu1 [11.2 kB]
Fetched 1,696 kB in 0s (21.2 MB/s)       
(Reading database ... 62741 files and directories currently installed.)
Removing scylla-server (0.10-1) ...
Selecting previously unselected package libaio1:amd64.
(Reading database ... 62720 files and directories currently installed.)
Preparing to unpack .../libaio1_0.3.109-4_amd64.deb ...
Unpacking libaio1:amd64 (0.3.109-4) ...
Selecting previously unselected package libboost-system1.55.0:amd64.
Preparing to unpack .../libboost-system1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-system1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-filesystem1.55.0:amd64.
Preparing to unpack .../libboost-filesystem1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-filesystem1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-program-options1.55.0:amd64.
Preparing to unpack .../libboost-program-options1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-program-options1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-test1.55.0:amd64.
Preparing to unpack .../libboost-test1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-test1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libboost-thread1.55.0:amd64.
Preparing to unpack .../libboost-thread1.55.0_1.55.0-1_amd64.deb ...
Unpacking libboost-thread1.55.0:amd64 (1.55.0-1) ...
Selecting previously unselected package libyaml-cpp0.5:amd64.
Preparing to unpack .../libyaml-cpp0.5_0.5.1-1_amd64.deb ...
Unpacking libyaml-cpp0.5:amd64 (0.5.1-1) ...
Selecting previously unselected package libjsoncpp0:amd64.
Preparing to unpack .../libjsoncpp0_0.6.0~rc2-3ubuntu1_amd64.deb ...
Unpacking libjsoncpp0:amd64 (0.6.0~rc2-3ubuntu1) ...
Selecting previously unselected package liblz4-1:amd64.
Preparing to unpack .../liblz4-1_0.0~r114-2ubuntu1_amd64.deb ...
Unpacking liblz4-1:amd64 (0.0~r114-2ubuntu1) ...
Selecting previously unselected package libcrypto++9.
Preparing to unpack .../libcrypto++9_5.6.1-6+deb8u1build0.14.04.1_amd64.deb ...
Unpacking libcrypto++9 (5.6.1-6+deb8u1build0.14.04.1) ...
Selecting previously unselected package libhugetlbfs0.
Preparing to unpack .../libhugetlbfs0_2.17-0ubuntu2_amd64.deb ...
Unpacking libhugetlbfs0 (2.17-0ubuntu2) ...
Selecting previously unselected package hugepages.
Preparing to unpack .../hugepages_2.17-0ubuntu2_amd64.deb ...
Unpacking hugepages (2.17-0ubuntu2) ...
Selecting previously unselected package libsnappy1.
Preparing to unpack .../libsnappy1_1.1.0-1ubuntu1_amd64.deb ...
Unpacking libsnappy1 (1.1.0-1ubuntu1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up libaio1:amd64 (0.3.109-4) ...
Setting up libboost-system1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-filesystem1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-program-options1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-test1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-thread1.55.0:amd64 (1.55.0-1) ...
Setting up libyaml-cpp0.5:amd64 (0.5.1-1) ...
Setting up libjsoncpp0:amd64 (0.6.0~rc2-3ubuntu1) ...
Setting up liblz4-1:amd64 (0.0~r114-2ubuntu1) ...
Setting up libcrypto++9 (5.6.1-6+deb8u1build0.14.04.1) ...
Setting up libhugetlbfs0 (2.17-0ubuntu2) ...
Setting up hugepages (2.17-0ubuntu2) ...
Setting up libsnappy1 (1.1.0-1ubuntu1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
ubuntu@ip-10-230-218-205:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
Selecting previously unselected package scylla-server.
(Reading database ... 62812 files and directories currently installed.)
Preparing to unpack scylla-server_0.10-1_amd64.deb ...
Unpacking scylla-server (0.10-1) ...
dpkg: dependency problems prevent configuration of scylla-server:
 scylla-server depends on libthrift0; however:
  Package libthrift0 is not installed.

dpkg: error processing package scylla-server (--install):
 dependency problems - leaving unconfigured
Processing triggers for ureadahead (0.100.0-16) ...
Errors were encountered while processing:
 scylla-server
ubuntu@ip-10-230-218-205:~$ 
ubuntu@ip-10-230-218-205:~$ sudo dpkg -i scylla-server_0.10-1_amd64.deb 
```

**Comment by syuu1228** [comment]
This is not a error, expected behavior.
You need to copy libthrift0_1.0.0-dev_amd64.deb and install it before installing scylla-server_0.10-1_amd64.deb.
When you create apt repository for scylla and install by "apt-get install scylla-server", it will automatically resolve.

**Comment by slivne** [comment]
Well 

I used current head

commit 3a3867588adea06244cdde26f3188e72c23d475e
Author: Vlad Zolotarov <vladz@cloudius-systems.com>
Date:   Tue Oct 27 12:38:13 2015 +0200

    tests::cql_test_env: set broadcast address
    
    cql_query_test hasn't configured Broadcast address before
    it was used for the first time.
    
    Broadcast address is an essential Node's configuration.
    
    There is an assert in utils::fb_utils::get_broadcast_address()
    that ensures that broadcast address has been properly configured
    before it's used for the first time and it is triggered without
    this patch.
    
    Signed-off-by: Vlad Zolotarov <vladz@cloudius-systems.com>

It does not build the thrift.deb  package (at least for me)

ubuntu@ip-10-148-3-46:~$ find . -name "*.deb"
./scylla-server_0.10-1_amd64.deb

sudo find / -name "*thrift*.deb" - does not find anything.

I guess something is broken - can you try to run a build on AWS ubuntu 1.4.04 c3.4X instance and see that it works and generates everything - it does not work for me.

does not find anything

**Comment by slivne** [comment]
@syuu1228 

Ok found it ... the thrift deb package is created with build_dependency and then installed - yet when we build scylla it seems they are erased as they are kept in build directory which is cleared at some point.

We need to copy them before they are overwritten - as you may have already multiple changes in flight - please include it in one of your patches.

**Comment by syuu1228** [comment]
I also realized someone erased build/ dir, but I'm not yet find who did it..

**Comment by slivne** [comment]
So lets work around it - lets copy the build/*.deb after build_depenency
and before the next steps.

**Comment by syuu1228** [comment]
Just sent a patch for this ( [PATCH scylla] dist: do not remove build/ dir when scylla-server ubuntu package building)

**Comment by slivne** [comment]
fixed by 5aab69e72f44ac67772c7c5a48bb9efdbdb42ddc
